server/metrics/metrics.go

```diff
@@ -2024,6 +2027,35 @@ var (
 	}, []string{
 		GroupID,
 	})
+
+	/// ## Podman metrics
+
+	PodmanSociStoreCrashes = promauto.NewCounter(prometheus.CounterOpts{
+		Namespace: bbNamespace,
+		Subsystem: "podman",
+		Name:      "soci_store_crash_count",
+		Help:      "Total number of times the soci store binary crashed and was restarted.",
+	})
+
+	PodmanGetSociArtifactsLatencyUsec = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: bbNamespace,
+		Subsystem: "podman",
+		Name:      "get_soci_artifacts_latency_usec",
+		Buckets:   durationUsecBuckets(1*time.Microsecond, 100*time.Minute, 10),
+		Help:      "The latency of retrieving SOCI artifacts from the app and storing them locally per image, in microseconds. Note this is slightly different than the latency of the GetArtifacts RPC as the artifacts must be fetched from the cache and stored locally, which adds some additional time.",
+	}, []string{
+		ContainerImageTag,
+	})
+
+	PodmanColdImagePullLatencyMsec = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: bbNamespace,
+		Subsystem: "podman",
+		Name:      "image_pull_latency_msec",
+		Buckets:   durationMsecBuckets(1*time.Millisecond, 100*time.Minute, 10),
+		Help:      "The latency of 'cold' podman pull requests per image, in milliseconds. 'Cold' means the image hasn't been pulled by this executor previously.",
+	}, []string{
+		ContainerImageTag,
```

Comment:
It's generally not advised to put dynamic data into labels since a basic prometheus setup will not really scale well to handle that. What are you trying to get out of this metric? i.e. what will knowing the cold pull time give you?

Reply:
Yeah, I saw that and chatted with Tyler about it today. We don't believe this data will be too problematic to use as a label (for now) as we don't expect there to be many more than ~100 images in use and we should remove this if that number changes significantly. I want to keep track of the image pull latency per-image to observe the effect image streaming has on image pull latency and keep an eye on it for a little while until we're comfortable everything is working correctly.

If you're super concerned about it I could instead hash and bucket the image tag into a fixed set of labels, though we would lose the ability to reverse from bucket to image tag in that case, which isn't the end of the world.